pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -102,14 +102,15 @@ func (p *LaunchTemplateProvider) Get(ctx context.Context, constraints *v1alpha1.
 	for amiID, instanceTypes := range amis {
 		// Get userData for Node
 		userData, err := p.getUserData(ctx, constraints, instanceTypes, additionalLabels)
+		instanceProfile := p.getInstanceProfile(ctx, constraints)
 		if err != nil {
 			return nil, err
 		}
 		// Ensure the launch template exists, or create it
 		launchTemplate, err := p.ensureLaunchTemplate(ctx, &launchTemplateOptions{
 			UserData:          userData,
 			ClusterName:       injection.GetOptions(ctx).ClusterName,
-			InstanceProfile:   constraints.InstanceProfile,
+			InstanceProfile:   instanceProfile,
```

Comment:
consider just inlining 
```
InstanceProfile: p.getInstanceProfile(ctx, constraints)
```

Reply:
Adjusted with latest commit, as the function now returns an error if no usable profile was found.